### ISSUE TITLE
Feishin stop opening after upgrade 0.5.1 -> 0.5.2 / 0.5.3 - Appimage - Ubuntu - Cinnamon

### ISSUE THREAD

**Issue by 7Adrian** [jeffvli/feishin#411]
## Expected Behavior

Feishin newest version opens.

## Current Behavior

Application shows in tray for a while and disappear, audio icon change for a while and back to "normal" state. Window never showing, even blinking.

Where I try to lunch version 0.5.2/0.5.3: 
<details>
  <summary>Click to expand logs</summary>

>```bash
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.003064:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.144485:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.280955:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.411515:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.540806:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[12707:1214/113852.669560:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=139
>: CommandLine Error: Option 'help-list' registered more than once!
>LLVM ERROR: inconsistency in registered CommandLine options
>[12707:1214/113852.847878:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=134
>: CommandLine Error: Option 'help-list' registered more than once!
>LLVM ERROR: inconsistency in registered CommandLine options
>[12707:1214/113853.002932:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=134
>: CommandLine Error: Option 'help-list' registered more than once!
>LLVM ERROR: inconsistency in registered CommandLine options
>[12707:1214/113853.166759:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=134
>[12707:1214/113853.166775:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.
>```
>
>App main.log:
>
>```
>[2023-12-14 12:40:37.354] [info]  Checking for update
>[2023-12-14 12:40:37.400] [info]  Generated new staging user ID: <MY_ID>
>```

</details>

To compare version 0.5.1:
<details>
  <summary>Click to expand logs</summary>

>```bash
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114029.443978:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114029.594685:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114029.721046:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114029.849154:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114029.977428:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>ac_compute_device_uuid's output is based on invalid pci bus info.
>[13155:1214/114030.104628:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=139
>Initializing MPV with data:  {
>```
>Later app show some errors in logs (verbose: ' MPV is not running')... but basically works after manually play song :)
>Errors example:
>```bash
>Error in main process Error: Socket is closed
>...
>
>MPV failed to set volume {
>  errcode: 8,
>  verbose: 'MPV is not running',
>...
>```
>But I think that errors aren't connected with current problem.
</details>

## Steps to Reproduce

1. Download appimage version 0.5.2/0.5.3 

2. Add executable permission `chmod +x`

3. Start appimage

## Possible Solution (Not obligatory)

- Running newest version in docker, and connect from browser works.

- I remove **feishin*** directory from **~/.cache/** and **~/.config** and try to run app, but doesn't help.

## Your Environment

-   Application version: `v0.5.2/v0.5.3`
-   Application executable: `Feishin-0.5.X-linux-x86_64.AppImage`
-   Operating System and version: `Ubuntu Cinnamon 23.10 x86_64 `
-   Server and version: `Navidrome v0.50.1`
-   Kernel: `6.5.0-14-generic`
-   DE: `Cinnamon`
-   WM: `Mutter (Muffin)`
-   WM Theme: `cinnamon (Default)`
-   Theme: `Adwaita [GTK2/3]`
-   Icons: `gnome [GTK2/3]`
-   CPU: `AMD Ryzen 9 5950X`
-   GPU: `AMD ATI Radeon RX 7900 XTX`

**Comment by jeffvli** [comment]
I'm not able to reproduce this. v0.5.3 AppImage opens fine for me when testing on PopOS 20.04.

The only thing that changed between v0.5.1 and v0.5.2+ is the electron version and electron-builder versions which were updated for security updates and NixOS support.

You could try building the binary yourself by:
- cloning the repo
- running `npm install`
- running `npm run package`
- using the appimage from `./release/build`

If the current configuration doesn't work, try bumping the electron and electron-builder versions to their latest e.g. electron v27.1.3 and electron-builder v24.9.1 and building like that too.

**Comment by 7Adrian** [comment]
Thanks for your answer. I tried using electron v27.1.3 and elctron-builder v24.9.1 and nothing changed, but when using old electron v25.8.1 and electron-builder v24.9.1 it works. When running a `npm install` I've got some dependency conflicts and need to run `npm install --force`.

<details>
  <summary>Click to expand logs</summary>

>```bash
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE could not resolve
> npm ERR! 
> npm ERR! While resolving: eslint-plugin-typescript-sort-keys@2.1.0
> npm ERR! Found: typescript@5.2.2
> npm ERR! node_modules/typescript
> npm ERR!   dev typescript@"^5.2.2" from the root project
> npm ERR!   peerOptional typescript@">=4.9.5" from cosmiconfig@8.3.6
> npm ERR!   node_modules/stylelint/node_modules/cosmiconfig
> npm ERR!     cosmiconfig@"^8.2.0" from stylelint@15.10.3
> npm ERR!     node_modules/stylelint
> npm ERR!       dev stylelint@"^15.10.3" from the root project
> npm ERR!       7 more (stylelint-config-css-modules, stylelint-scss, ...)
> npm ERR!   4 more (ts-loader, ts-node, tsutils, typescript-plugin-styled-components)
> npm ERR! 
> npm ERR! Could not resolve dependency:
> npm ERR! peer typescript@"^3 || ^4" from eslint-plugin-typescript-sort-keys@2.1.0
> npm ERR! node_modules/eslint-plugin-typescript-sort-keys
> npm ERR!   dev eslint-plugin-typescript-sort-keys@"^2.1.0" from the root project
> npm ERR! 
> npm ERR! Conflicting peer dependency: typescript@4.9.5
> npm ERR! node_modules/typescript
> npm ERR!   peer typescript@"^3 || ^4" from eslint-plugin-typescript-sort-keys@2.1.0
> npm ERR!   node_modules/eslint-plugin-typescript-sort-keys
> npm ERR!     dev eslint-plugin-typescript-sort-keys@"^2.1.0" from the root project
> npm ERR! 
> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force, or --legacy-peer-deps
> npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
>
>```
</details>

I tried to find which version of electron cause errors and last working version is `v26.6.2`. In my case version `v27.0.0` and later cause that issue. Tomorrow I will try to check it in virtual machine with Ubuntu and Linux Mint to see if that problem exists.

**Comment by jeffvli** [comment]
Did you already test on Electron v28.0.0? That's the latest stable release and doesn't have any breaking changes that affect Feishin from what I can see. If it works in v28 I can bump it to there, or as a last resort scenario I can just create a branch that releases a binary with electron v25/v26.

**Comment by 7Adrian** [comment]
I tried v28.0.0 but received an error during compilation. 

I tested your version on **Linux Mint** in a VM and on a fresh **Ubuntu** installation in a VM, and in both cases, the **Feishin v0.5.3 AppImage worked**. So, I believe the issue is specific to my setup, and I must have messed something up. I've been using this system for years, upgrading, migrating between newer machines/technologies, learning, testing, etc.

I think we can close this issue. Thanks for your help! 😃
